### PR TITLE
feat: [CDS-44046]:link-templates

### PR DESCRIPTION
**Pull request by litvinovstas**
### Summary
add accountId parameter for yaml snippets api

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by litvinovstas** [comment]
trigger cypress

File: src/modules/70-pipeline/components/PipelineStudio/ExecutionStrategy/__tests__/Phases.test.tsx [hunk 1]
```diff
@@ -156,47 +147,69 @@ describe('Phases test', () => {
...
+    const { container } = render(
+      <WrapperComponent
+        context={pipelineContextMockValue}
+        selectedStrategy={'BlankCanvas'}
+        serviceDefinitionType={jest.fn().mockReturnValue('WinRm')}
+        initialValues={{}}
+      />
+    )
+
+    expect(container).toBeDefined()
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/components/PipelineStudio/ExecutionStrategy/__tests__/Phases.test.tsx:173]: Check for some DOM element, even `null` returns true when checked against `toBeDefined`

File: src/modules/70-pipeline/components/PipelineStudio/ExecutionStrategy/__tests__/Phases.test.tsx [hunk 2]
```diff
@@ -156,47 +147,69 @@ describe('Phases test', () => {
...
+    expect(switchType).toBeInTheDocument()
+    fireEvent.click(switchType)
+    const percentageOption = document.body.querySelector('a[data-name="percentage"]') as HTMLElement
+    expect(percentageOption).toBeInTheDocument()
+    fireEvent.click(percentageOption)
+    const inputs = container.querySelectorAll('.bp3-input')
+    fireEvent.change(inputs[1], { target: { value: 90 } })
+    fireEvent.change(inputs[2], { target: { value: 99 } })
+
+    expect(container).toBeDefined()
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/components/PipelineStudio/ExecutionStrategy/__tests__/Phases.test.tsx:208]: Same here. 
And we have found may elements above from container, if those assertions are passing above then this assertion is not required or not adding much value.

File: src/modules/70-pipeline/components/PipelineStudio/ExecutionStrategy/__tests__/Phases.test.tsx [hunk 3]
```diff
@@ -156,47 +147,69 @@ describe('Phases test', () => {
...
+    await act(async () => {
+      clickSubmit(container)
+    })
+
+    await waitFor(() => expect(pipelineContextMockValue.updateStage).toHaveBeenCalled())
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/components/PipelineStudio/ExecutionStrategy/__tests__/Phases.test.tsx:213]: User `toHaveBeenCallWith` here and check if correct argument is getting passed to updateStage

**Review by chetanvekariya-harness — APPROVED** [review]

**Review by acasian-harness — APPROVED** [review]

**Review by vikrant-harness — APPROVED** [review]

**Review by bhavya-sinha — APPROVED** [review]

**Review by vkbansal-harness — APPROVED** [review]

**Pull request merged**